evap/staff/tests/test_forms.py

```diff
@@ -597,3 +594,31 @@ def test_voter_cannot_be_removed_from_evaluation(self):
         self.assertFalse(form.is_valid())
         self.assertIn('participants', form.errors)
         self.assertIn("Participants who already voted for the evaluation can't be removed", form.errors['participants'][0])
+
+    def test_course_change_updates_cache(self):
+        semester = mommy.make(Semester)
+        course1 = mommy.make(Course, semester=semester)
+        course2 = mommy.make(Course, semester=semester)
+        evaluation = mommy.make(Evaluation, course=course1)
+        evaluation.general_contribution.questionnaires.set([mommy.make(Questionnaire)])
+
+        # save without changes
+        form_data = get_form_data_from_instance(EvaluationForm, evaluation)
+        form = EvaluationForm(form_data, instance=evaluation, semester=semester)
+        self.assertTrue(form.is_valid())
+        with patch('evap.results.views._delete_course_template_cache_impl') as delete_call, patch('evap.results.views.warm_up_template_cache') as warmup_call:
+            form.save()
+        self.assertEqual(Evaluation.objects.get(pk=evaluation.pk).course, course1)
+        self.assertEqual(delete_call.call_count, 0)
+        self.assertEqual(warmup_call.call_count, 0)
+
+        # change course and save
+        form_data = get_form_data_from_instance(EvaluationForm, evaluation)
+        form_data["course"] = course2.pk
+        form = EvaluationForm(form_data, instance=evaluation, semester=semester)
+        self.assertTrue(form.is_valid())
+        with patch('evap.results.views._delete_course_template_cache_impl') as delete_call, patch('evap.results.views.warm_up_template_cache') as warmup_call:
```

Comment:
Unify this with the patch above to prevent the first assertions just being true due to typing errors, please.